src/lua/uri.lua

```diff
@@ -60,6 +60,41 @@ local uri_set_stash = buffer.ffi_stash_new('struct uri_set')
 local uri_set_stash_take = uri_set_stash.take
 local uri_set_stash_put = uri_set_stash.put
 
+-- RFC 3986: Uniform Resource Identifier (URI): Generic Syntax
+-- 2.2. Reserved Characters
+-- https://datatracker.ietf.org/doc/html/rfc3986.html#section-2.2
+local UNRESERVED_RFC3986 = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-._~"
+
+local RFC3986 = {
+    unreserved = UNRESERVED_RFC3986,
+    plus = false,
+}
+
+local FORM_DATA = {
+    unreserved = UNRESERVED_RFC3986,
+    plus = true,
+}
+
+local PATH_SEGMENT = {
+    unreserved = UNRESERVED_RFC3986 .. ":@-._~!$&'()*+,;=",
+    plus = false,
+}
+
+local PATH = {
+    unreserved = UNRESERVED_RFC3986 .. ":@-._~!$&'()*+,;=/",
+    plus = false,
+}
+
+local QUERY = {
+    unreserved = UNRESERVED_RFC3986 .. "/",
+    plus = false,
+}
+
+local FRAGMENT = {
+    unreserved = UNRESERVED_RFC3986 .. "/?:@-._~!$&'()*+,;=",
+    plus = false,
+}
```

Comment:
We should either describe a logic behind those set of characters or link a source.

Reply:
It is what I'll review disregarding of whether it'll go into the documentation or not. Reopened.